spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbAutoConfigurationTest.java

```diff
@@ -0,0 +1,135 @@
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.utils.AttributeMap;
+
+/**
+ *
+ * @author Matej Nedic
+ */
+public class DynamoDbAutoConfigurationTest {
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withPropertyValues("spring.cloud.aws.region.static:eu-west-1")
+			.withConfiguration(AutoConfigurations.of(RegionProviderAutoConfiguration.class,
+					CredentialsProviderAutoConfiguration.class, DynamoDbAutoConfiguration.class,
+					AwsAutoConfiguration.class));
+
+	@Test
+	void dynamoDBAutoConfigurationIsDisabled() {
+		this.contextRunner.withPropertyValues("spring.cloud.aws.dynamodb.enabled:false")
+				.run(context -> assertThat(context).doesNotHaveBean(DynamoDbClient.class));
+	}
+
+	@Test
+	void dynamoDBAutoConfigurationIsEnabled() {
+		this.contextRunner.withPropertyValues("spring.cloud.aws.dynamodb.enabled:true").run(context -> {
+			assertThat(context).hasSingleBean(DynamoDbClient.class);
+			assertThat(context).hasSingleBean(DynamoDbTemplate.class);
+			assertThat(context).hasSingleBean(DynamoDbEnhancedClient.class);
+
+			DynamoDbClient client = context.getBean(DynamoDbClient.class);
+			SdkClientConfiguration clientConfiguration = (SdkClientConfiguration) ReflectionTestUtils.getField(client,
+					"clientConfiguration");
+			AttributeMap attributes = (AttributeMap) ReflectionTestUtils.getField(clientConfiguration, "attributes");
+			assertThat(attributes.get(SdkClientOption.ENDPOINT))
+					.isEqualTo(URI.create("https://dynamodb.eu-west-1.amazonaws.com"));
+
+		});
+	}
+
+	@Test
+	void withCustomEndpoint() {
+		this.contextRunner.withPropertyValues("spring.cloud.aws.dynamodb.endpoint:http://localhost:8090")
+				.run(context -> {
+					assertThat(context).hasSingleBean(DynamoDbClient.class);
+					assertThat(context).hasSingleBean(DynamoDbTemplate.class);
+					assertThat(context).hasSingleBean(DynamoDbEnhancedClient.class);
+
+					DynamoDbClient client = context.getBean(DynamoDbClient.class);
+
+					SdkClientConfiguration clientConfiguration = (SdkClientConfiguration) ReflectionTestUtils
+							.getField(client, "clientConfiguration");
+					AttributeMap attributes = (AttributeMap) ReflectionTestUtils.getField(clientConfiguration,
+							"attributes");
+					assertThat(attributes.get(SdkClientOption.ENDPOINT)).isEqualTo(URI.create("http://localhost:8090"));
+					assertThat(attributes.get(SdkClientOption.ENDPOINT_OVERRIDDEN)).isTrue();
+				});
+	}
+
+	@Test
+	void customTableResolverResolverCanBeConfigured() {
+		this.contextRunner.withUserConfiguration(DynamoDbAutoConfigurationTest.CustomDynamoDbConfiguration.class)
+				.run(context -> {
+					assertThat(context).hasSingleBean(DynamoDbAutoConfigurationTest.CustomDynamoDBTableResolver.class);
```

Comment:
We should check here that there is a single bean of type `DynamoDbTableResolver` and it is an instance of `CustomDynamoDBTableResolver`